app/features/notes_generator/core.py

```diff
@@ -0,0 +1,64 @@
+import re
+from typing import Literal, Optional, Dict
+import logging
+from tools import (
+    extract_text_from_file,
+    extract_text_from_url,
+    generate_notes_rag,
+    clean_content
+)
+
+# Dictionary to map input types to corresponding functions
+file_handler_map = {
+    'text': lambda content: clean_content(content),  # Clean content directly
+    'file': extract_text_from_file,
+    'url': extract_text_from_url
+}
+
+# List of supported file extensions
+SUPPORTED_FILE_TYPES = ['.txt', '.docx', '.pdf', '.csv', '.pptx', '.md', '.xml', '.xls', '.xlsx']
```

Comment:
These are not all the supported file types from our Marvel AI Backend. You can check the `utils/document_loaders.py` file for knowing which file types are supported for our system.